R/setup.R

```diff
@@ -249,10 +244,7 @@ Leaf_setup <- function(aLandAllocator, aRegionName, aData, aColName,
 
       # Check whether leaf is a new technology
       if(!is.null(newTechData)) {
-        newTechData %>%
-          filter(LandLeaf == childName) ->
-          tempNewTech
-        if(nrow(tempNewTech)) {
+         if(childName %in% unique(newTechData$LandLeaf)) {
```

Comment:
You probably don't need the call to `unique` here.  If `childName` is in the unique'd version, then it is in the original, and the direct search should be faster:
```R
v <- sample.int(100, 1000, TRUE)
microbenchmark::microbenchmark(13 %in% v, 13 %in% unique(v))
```
```
Unit: microseconds
              expr    min      lq     mean  median      uq    max neval
         13 %in% v  2.833  3.2230  8.82786  3.7925 13.0235 80.552   100
 13 %in% unique(v) 16.137 17.1625 27.57659 26.6350 32.8425 88.099   100
```

```suggestion
         if(childName %in% newTechData$LandLeaf) {
```

Reply:
BTW, I just reprofiled this branch, and a 4-period run takes about 18 s, of which about 5.4 s is being attributed to `unique`.  Some of those are being called from `lm`, but it looks like most are being called from `%in%`, so it's not huge, but it's not negligible.  

That said, some of that time is coming from `%in% unique(...)` constructs in the setup.  The single largest (timewise) example appears to be this one, which gets executed only once, but takes 150ms all by itself.
```R
    # Set cost in the LandLeaf
    if(aScenarioInfo$mUseZeroCost == FALSE &
       name %in% unique(cost$AgProductionTechnology) & y %in% unique(cost$year))  {
      aLandLeaf$mCost[per] <- as.numeric(cost$nonLandVariableCost[cost$year == y &
                                                                    cost$AgProductionTechnology == name])
    } else {
      aLandLeaf$mCost[per] <- 0
    }
```